test/requests/hover_expectations_test.rb

```diff
@@ -61,6 +61,38 @@ class A
     T.must(message_queue).close
   end
 
+  def test_hovering_methods_defined_on_self
+    message_queue = Thread::Queue.new
+    store = RubyLsp::Store.new
+
+    uri = URI("file:///fake.rb")
+    source = <<~RUBY
+      class A
+        # Hello from `foo`
+        def foo; end
+
+        def bar
+          foo
```

Comment:
Let's add `self.foo` as well.

Reply:
Let's please add singleton methods after #1113. There are a few refactors we need to do for that PR that will influence the implementation (e.g.: getting rid of the `SingletonMethod` class and just storing the right owner).

Reply:
Oh, sorry, I got confused. Added the test cases for explicit `self.foo`.